lib/inv_pcg_quda.cpp

```diff
@@ -134,13 +155,14 @@ namespace quda
 
     if (K) {
       csParam.create = QUDA_COPY_FIELD_CREATE;
-      csParam.setPrecision(param.precision_precondition);
+      csParam.setPrecision(Kparam.precision);
+#ifdef FLOAT8
+      csParam.fieldOrder = QUDA_FLOAT8_FIELD_ORDER;
```

Comment:
Is this safe for all possible preconditioners?

Reply:
No and that's intentional.  As noted above, `FLOAT8` support is a known wart since it is a non-native type which will require some thought if and how to deploy en masse in QUDA.  There's way to determine if the preconditioner is safe to use with `FLOAT8` mode, but I will add a warning at this point to let the user know they are using an experimental feature.

Reply:
@maddyscientist